### ISSUE TITLE
compiler warning in FrontendDialectHelper.cpp

### ISSUE THREAD

**Issue by AlexandreEichenberger** [onnx/onnx-mlir#1136]
a recent commit introduced this warning

```
/workdir/onnx-mlir/src/Builder/FrontendDialectHelper.cpp: In function 'mlir::Type onnx_mlir::convertONNXTypeToMLIRType(mlir::OpBuilder&, onnx::TensorProto_DataType)':
/workdir/onnx-mlir/src/Builder/FrontendDialectHelper.cpp:241:10: warning: enumeration value 'TensorProto_DataType_BFLOAT16' not handled in switch [-Wswitch]
  241 |   switch (onnxType) {
      |          ^
```

The proper case should be added, or a default with llvm-unreachable.

**Comment by AlexandreEichenberger** [comment]
@chentong319 do you mind to have a look? Thanks

**Comment by chentong319** [comment]
@AlexandreEichenberger I pushed the fix with PR#1168.
